torch/CMakeLists.txt

```diff
@@ -82,6 +82,8 @@ set(TORCH_PYTHON_SRCS
     ${TORCH_SRC_DIR}/csrc/jit/script/python_tree_views.cpp
     ${TORCH_SRC_DIR}/csrc/multiprocessing/init.cpp
     ${TORCH_SRC_DIR}/csrc/onnx/init.cpp
+    ${TORCH_SRC_DIR}/csrc/perf_tools/init.cpp
+    ${TORCH_SRC_DIR}/csrc/perf_tools/ThroughputBenchmark.cpp
```

Comment:
more bike-shedding, `csrc/utils/throughput_benchmark.cpp` to make it consistent with `torch.utils` on the Python side? (more of a soft question)

Reply:
@soumith , sure, consistency is good and we should keep things consistent. 
One issue with "utils" name is that it becomes a bug of stuff. I guess it can be a smaller issue with PyTorch given extensive reviews we have. But in Facebook such general things usually end up being a mess.

What do you think, do you prefer to keep torch.utils.ThroughputBenchmark? Another option I see is to create a new module called torch.perf_tools. This also can motivate people to contribute more to the space :)

Reply:
i think we are and will be diligent about code dumping. definitely dont want it to become a mess.

once we actually think through how to extend this utility to be a more general throughput / serving simulator (for example serving GPU or other accelerators, what the API would look like to generalize over -- what would serving look like via multiprocessing), i think we should definitely open up a `torch.profiling` namespace. I am just wary of opening up the namespace now with this single utility in it's current form.

Reply:
Fair enough, I like this approach!